dex/testing/dcr/harness.sh

```diff
@@ -0,0 +1,242 @@
+  case \$1 in
+      ''|*[!0-9]*)  ;;
+      *) NUM=\$1 ;;
+  esac
+  for i in \$(seq \$NUM) ; do
+    dcrctl -C ${NODES_ROOT}/beta/beta-ctl.conf generate 1
+    sleep 0.5
+  done
+EOF
+chmod +x "${NODES_ROOT}/harness-ctl/mine-beta"
+
+# Reorg script
+cat > "${NODES_ROOT}/harness-ctl/reorg" <<EOF
+#!/bin/sh
+echo "Disconnecting beta from alpha"
+sleep 1
+./beta addnode 127.0.0.1:${ALPHA_PORT} remove
+echo "Mining a block on alpha"
+sleep 1
+./mine-alpha 1
+echo "Mining 3 blocks on beta"
+./mine-beta 3
+sleep 2
+echo "Reconnecting beta to alpha"
+./beta addnode 127.0.0.1:${ALPHA_PORT} add
+sleep 2
+grep REORG ${NODES_ROOT}/alpha/logs/simnet/dcrd.log
+EOF
+chmod +x "${NODES_ROOT}/harness-ctl/reorg"
+
+# Shutdown script
+cat > "${NODES_ROOT}/harness-ctl/quit" <<EOF
+#!/bin/sh
+tmux send-keys -t $SESSION:0 C-c
+tmux send-keys -t $SESSION:1 C-c
+tmux wait-for alpha
+tmux wait-for beta
+tmux kill-session
+EOF
+chmod +x "${NODES_ROOT}/harness-ctl/quit"
+
+################################################################################
+# dcrd Nodes
+################################################################################
+
+cd ${NODES_ROOT} && tmux new-session -d -s $SESSION
+
+tmux rename-window -t $SESSION:0 'alpha'
+tmux send-keys "cd ${NODES_ROOT}/alpha" C-m
+
+echo "Starting simnet alpha node"
+tmux send-keys "dcrd --appdata=${NODES_ROOT}/alpha \
+--rpcuser=${RPC_USER} --rpcpass=${RPC_PASS} \
+--miningaddr=${ALPHA_MINING_ADDR} --rpclisten=127.0.0.1:${ALPHA_RPC_PORT} \
+--txindex --listen=127.0.0.1:${ALPHA_PORT} \
+--debuglevel=debug \
+--simnet; tmux wait-for -S alpha" C-m
+
+tmux new-window -t $SESSION:1 -n 'beta'
+tmux send-keys "cd ${NODES_ROOT}/beta" C-m
+
+echo "Starting simnet beta node"
+tmux send-keys "dcrd --appdata=${NODES_ROOT}/beta \
+--rpcuser=${RPC_USER} --rpcpass=${RPC_PASS} \
+--listen=127.0.0.1:19559 --rpclisten=127.0.0.1:${BETA_RPC_PORT} \
+--miningaddr=${BETA_MINING_ADDR} \
+--txindex --connect=127.0.0.1:${ALPHA_PORT} \
+--debuglevel=debug \
+--simnet; tmux wait-for -S beta" C-m
+
+sleep 3
+
+################################################################################
+# dcrwallets
+################################################################################
+
+tmux new-window -t $SESSION:2 -n 'w-alpha'
+tmux send-keys "cd ${NODES_ROOT}/alpha" C-m
+tmux send-keys "dcrwallet -C w-alpha.conf --create" C-m
+echo "Creating simnet alpha wallet"
+sleep 1
+tmux send-keys "${WALLET_PASS}" C-m "${WALLET_PASS}" C-m "n" C-m "y" C-m
+sleep 1
+tmux send-keys "${ALPHA_WALLET_SEED}" C-m C-m
+tmux send-keys "dcrwallet -C w-alpha.conf " C-m
+
+tmux new-window -t $SESSION:3 -n 'w-beta'
+tmux send-keys "cd ${NODES_ROOT}/beta" C-m
+tmux send-keys "dcrwallet -C w-beta.conf --create" C-m
+echo "Creating simnet beta wallet"
+sleep 1
+tmux send-keys "${WALLET_PASS}" C-m "${WALLET_PASS}" C-m "n" C-m "y" C-m
+sleep 1
+tmux send-keys "${BETA_WALLET_SEED}" C-m C-m
+tmux send-keys "dcrwallet -C w-beta.conf --debuglevel=debug" C-m
+
+sleep 30
```

Comment:
Here's another place that is potentially indicative of some underlying issue in either dcrwallet or dcrd. If this wait is not long enough, the script sometimes hangs at the first call to `sendtoaddress` below.